### ISSUE TITLE
Extend dragAndDrop command to allow offset values

### ISSUE THREAD

**Issue by just-boris** [webdriverio/webdriverio#4916]
**Is your feature request related to a problem? Please describe.**
Currently, dragAndDrop command allows only to move one element to another

```js
$(selector).dragAndDrop(target, duration)
```

There is no way to move a single element to a certain offset

**Describe the solution you'd like**

Create a new command `dragTo`, which performs the same actions, but allows to specify offset, instead of a target selector 

```js
$(selector).dragTo(xoffset, yoffset, duration)
```

**Describe alternatives you've considered**
Currently I can implement this myself, using `performActions` command, but this is not cross-browser and I had to maintain w3c/jsonwp implementations, which is cumbersome. 

Having this in the webdriverio library might be useful for somebody.

**Additional context**
I am testing a slider component, which requires dragging interactions, but there is no target element, the slider itself is being dragged. The proposed command could resolve my use-case

**Comment by erwinheitzman** [comment]
I like this proposal but would like to change this after we release v6 as we are planning to change some commands to accept option objects instead which would allow for more clearance and better usability :)

**Comment by unickq** [comment]
@just-boris 

Can you share your examples that work?

**Comment by just-boris** [comment]
Here is our custom implementation

```js
const ACTION_BUTTON = 0;
const dragDuration = 100;

async function dragAndDrop(browser, sourceSelector, targetX = 0, targetY = 0) {
        if (!this.browser.isW3C) {
            const element = await this.browser.$(sourceSelector);
            await element.moveTo();
            await this.browser.buttonDown(ACTION_BUTTON);
            await element.moveTo(targetX, targetY);
            await this.browser.buttonUp(ACTION_BUTTON);
            return;
        }
        const box = await this.getBoundingBox(sourceSelector);
        const center = getElementCenter(box);
        // W3C way of handle the drag and drop action
        await this.browser.performActions([
            {
                type: 'pointer',
                id: 'mouse',
                parameters: { pointerType: 'mouse' },
                actions: [
                    { type: 'pointerMove', duration: 0, x: center.x, y: center.y },
                    { type: 'pointerDown', button: ACTION_BUTTON },
                    { type: 'pointerMove', duration: dragDuration, origin: 'pointer', x: targetX, y: targetY },
                    { type: 'pointerUp', button: ACTION_BUTTON }
                ]
            }
        ]);
        // make sure all controls are properly released to avoid conflicts with further actions
        await this.browser.releaseActions();
    }
```

**Comment by unickq** [comment]
@just-boris 

Thanks. Worked like a charm :)

Just an example for guys who use wdio sync and TS
``` typescript
export function dragAndDrop(
  element: WebdriverIO.Element,
  targetX: number = 0,
  targetY: number = 0
) {
  const ACTION_BUTTON = 0;
  const dragDuration = 100;
  const center = element.getLocation();

  browser.performActions([
    {
      type: "pointer",
      id: "mouse",
      parameters: { pointerType: "mouse" },
      actions: [
        {
          type: "pointerMove",
          duration: 0,
          x: Math.round(center.x),
          y: Math.round(center.y)
        },
        { type: "pointerDown", button: ACTION_BUTTON },
        {
          type: "pointerMove",
          duration: dragDuration,
          origin: "pointer",
          x: targetX,
          y: targetY
        },
        { type: "pointerUp", button: ACTION_BUTTON }
      ]
    }
  ]);
  browser.releaseActions();
}
```

**Comment by christian-bromann** [comment]
Fixed in #5282
